crates/e2e-test-utils/src/node.rs

```diff
@@ -48,48 +52,76 @@ where
                 _marker: PhantomData::<Node::Engine>,
             },
             rpc: RpcTestContext { inner: node.rpc_registry },
+            wallets: WalletGenerator::new(2).chain_id(MAINNET.chain).gen(),
         })
     }
 
-    pub async fn connect(&mut self, node: &mut NodeTestContext<Node>) {
-        self.network.add_peer(node.network.record()).await;
-        node.network.add_peer(self.network.record()).await;
-        node.network.expect_session().await;
-        self.network.expect_session().await;
+    /// Overrides the default wallets with a given amount of wallets and its chain id
+    pub fn with_wallets(mut self, chain_id: ChainId, amount: usize) -> Self {
+        self.wallets = WalletGenerator::new(amount).chain_id(chain_id).gen();
+        self
     }
 
     /// Advances the chain `length` blocks.
     ///
     /// Returns the added chain as a Vec of block hashes.
-    pub async fn advance(
+    pub async fn advance_many(
         &mut self,
         length: u64,
-        tx_generator: impl Fn(u64) -> Pin<Box<dyn Future<Output = Bytes>>>,
+        tx_generator: impl Fn(u64) -> Pin<Box<dyn Future<Output = Bytes> + Send>>,
         attributes_generator: impl Fn(u64) -> <Node::Engine as EngineTypes>::PayloadBuilderAttributes
             + Copy,
-    ) -> eyre::Result<
-        Vec<(
-            <Node::Engine as EngineTypes>::BuiltPayload,
-            <Node::Engine as EngineTypes>::PayloadBuilderAttributes,
-        )>,
-    >
+    ) -> eyre::Result<Vec<B256>>
```

Comment:
having a chain of payloads is useful when creating sidechains, and wanting to rely on submit payload endpoints